tests/unit/lifecycle/LifecycleRetriggerRestoreTask.spec.js

```diff
@@ -0,0 +1,90 @@
+const assert = require('assert');
+const werelogs = require('werelogs');
+
+const { ObjectMD, ObjectMDArchive } = require('arsenal').models;
+const ActionQueueEntry = require('../../../lib/models/ActionQueueEntry');
+const { LifecycleRetriggerRestoreTask } = require(
+    '../../../extensions/lifecycle/tasks/LifecycleRetriggerRestoreTask');
+
+const {
+    BackbeatMetadataProxyMock,
+    ProcessorMock,
+} = require('../mocks');
+
+describe('LifecycleResetTransitionInProgressTask', () => {
+    let backbeatMetadataProxyClient;
+    let objectProcessor;
+    let task;
+
+    const actionEntry = ActionQueueEntry.create('requeueTransition')
+        .setAttribute('target', {
+            byAccount: {
+                123: {
+                    bucket1: [{
+                        objectKey: 'obj1',
+                        objectVersion: 'v1',
+                        try: 12,
+                    }]
+                },
+            },
+        });
+
+    const objectNotArchived = new ObjectMD();
+    const archiveInfo = new ObjectMDArchive({
+        archiveId: '123456789'
+    }, Date.now(), 1, undefined, undefined);
+    const objectArchived = new ObjectMD()
+        .setArchive(archiveInfo)
+        .setUserMetadata({
+            'x-amz-meta-scal-s3-restore-attempt': 11,
+        });
+
+    beforeEach(() => {
+        backbeatMetadataProxyClient = new BackbeatMetadataProxyMock();
+
+        objectProcessor = new ProcessorMock(
+            null,
+            null,
+            backbeatMetadataProxyClient,
+            null,
+            new werelogs.Logger('test:LifecycleRetriggerRestoreTask'));
+
+        task = new LifecycleRetriggerRestoreTask(objectProcessor);
+    });
+
+    it('should skip object not archived', done => {
```

Comment:
missing test cases for the various conditions to skip/!skip : already restored, restored but expired, restore not requested, ...